File: left-and-right-sum-differences/left-and-right-sum-differences.cpp
class Solution {
public:
  vector<int> leftRightDifference(vector<int>& nums) {
    int leftSum = 0, rightSum = 0;
    for ( int i = nums.size() - 1; i > 0; i-- ) {
      rightSum += nums[i];
    }

    vector < int > result;
    result.push_back(abs(leftSum - rightSum));
    for ( int i = 0; i < nums.size() - 1; i++ ) {
      leftSum += nums[i];
      rightSum -= nums[i+1];
      result.push_back(abs(leftSum - rightSum));
    }

    return result;
  }
};

File: left-and-right-sum-differences/left-and-right-sum-differences.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var leftRightDifference = function(nums) {
  let leftSum = 0, rightSum = 0
  for ( let i = nums.length - 1; i > 0; i-- ) {
    rightSum += nums[i]
  }

  let result = []
  result.push(Math.abs(leftSum - rightSum))
  for ( let i = 0; i < nums.length - 1; i++ ) {
    leftSum += nums[i]
    rightSum -= nums[i+1]
    result.push(Math.abs(leftSum - rightSum))
  }

  return result
};